### PR TITLE
hredis: cleanup, documentation

### LINKED ISSUE #34: hredis: github.com/garyburd/redigo is moved to github.com/gomodule/redigo
Future development of Redigo is at https://github.com/gomodule/redigo.

### PR DESCRIPTION
**Pull request by Xe**
- [x] Documentation
  - [x] functions
  - [x] package commentary
  - [x] license comments
- [x] Test everything
  - [x] Autostart redis on travis boot
- [x] Examples
- [x] Fix #34
- [x] Use [redcon](https://github.com/tidwall/redcon) for automated redis-less testing

### CODE REVIEW

File: hredis/doc.go [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/* Copyright (c) 2018 Salesforce
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root  or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+// Package hredis is a small set of utilities and useful functions for dealing
```
Comment (freeformz) [inline hredis/doc.go]: What do you think about this instead?
```go
Package hredis contains functions that are useful when using Heroku's Redis Addons.
```

File: hredis/example_test.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+		log.Fatal("can't connect to redis due to a timeout, check connection?")
+	}
+}
+
+func ExampleNewRedisPoolFromURL() {
+	p, err := NewRedisPoolFromURL(os.Getenv("REDIS_URL"))
+	if err != nil {
+		log.Fatal(err)
+	}
+
```
Comment (freeformz) [inline hredis/example_test.go]: What do you think about?
```go
defer p.Close()
// do something with pool 'p'
```

File: hredis/hredis_test.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	}{
+		{url: "redis://ad:hunter2@127.0.0.1:80", shouldErr: false},
+		{url: "rediss://ad:hunter2@127.0.0.1:80", shouldErr: false},
+		{url: "http://google.com", shouldErr: true},
+		{url: "redis://ad:hunter2@127.0.0.1:port", shouldErr: true},
+	}
+
+	for _, cs := range cases {
+		t.Run(cs.url, func(t *testing.T) {
+			_, err := RedissURL(cs.url)
```
Comment (freeformz) [inline hredis/hredis_test.go]: How about a test for the resulting url?

File: hredis/redigo_test.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import (
+	"testing"
+	"time"
+)
+
+// dontWait for time to pass, we're a TARDIS or something.
+func dontWait(t time.Time) error { return nil }
+
+func TestWaitForAvailability(t *testing.T) {
+	ok, err := WaitForAvailability("redis://127.0.0.1:6379", time.Millisecond, dontWait)
```
Comment (freeformz) [inline hredis/redigo_test.go]: What can we do to not require a local redis server for this test and the one below?

File: hredis/hredis_test.go [hunk 5]
```diff
@@ -30,6 +33,11 @@ func TestRedissURL(t *testing.T) {
 			if err != nil && !cs.shouldErr {
 				t.Fatalf("wanted nil error but got: %v", err)
 			}
+
+			_, err = url.Parse(u)
+			if err != nil {
+				t.Fatal(err)
+			}
```
Comment (freeformz) [inline hredis/hredis_test.go]: Given each test case what do you expect scheme to be?

Comment (Xe) [reply hredis/hredis_test.go]: Fixed in https://github.com/heroku/x/pull/32/commits/e7c2058bd7518a2b4fbef0f90815bc87524b9b05

File: .travis.yml [hunk 6]
```diff
@@ -3,4 +3,6 @@ go: '1.10'
 sudo: false
 install: true
 script:
-- go test -v -race ./...
+  - go test -v -race ./...
+services:
```
Comment (freeformz) [inline .travis.yml]: If we're using redcon, why do we need this?

**Comment by freeformz** [comment]
Please fix the tests though.

**Comment by jheth** [comment]
@Xe @freeformz   Do you see this PR landing soon? I'm interested in getting the change to gomodule/redigo.

**Comment by apghero** [comment]
This is pretty stale. Closing.

**Pull request closed without merging**